dask/array/slicing.py

```diff
@@ -35,28 +47,33 @@ def sanitize_index(ind):
     ...
     IndexError: Bad index.  Must be integer-like: 0.5
     """
-    if isinstance(ind, Number):
-        ind2 = int(ind)
-        if ind2 != ind:
-            raise IndexError("Bad index.  Must be integer-like: %s" % ind)
-        else:
-            return ind2
-    if hasattr(ind, 'tolist'):
-        ind = ind.tolist()
-    if isinstance(ind, list) and ind and isinstance(ind[0], bool):
-        ind = [a for a, b in enumerate(ind) if b]
-        return ind
-    if isinstance(ind, list):
-        return [sanitize_index(i) for i in ind]
-    if isinstance(ind, slice):
-        return slice(sanitize_index(ind.start),
-                     sanitize_index(ind.stop),
-                     sanitize_index(ind.step))
     if ind is None:
-        return ind
-    try:
-        return sanitize_index(np.array(ind).tolist())
-    except:
+        return None
+    elif isinstance(ind, slice):
+        return slice(_sanitize_index_element(ind.start),
+                     _sanitize_index_element(ind.stop),
+                     _sanitize_index_element(ind.step))
+    elif isinstance(ind, Number):
+        return _sanitize_index_element(ind)
+    index_array = np.asanyarray(ind)
+    if index_array.dtype == bool:
+        nonzero = np.nonzero(index_array)
+        if len(nonzero) == 1:
```

Comment:
I'm not sure about this line. Is it necessary for a given special case? Perhaps adding a comment would be useful.